### PR TITLE
TLPORTAL-213, TLPORTAL-214: improve handlng of incomplete data from ESB

### PR DESCRIPTION
**Pull request by dlhaines**

### CODE REVIEW

**Comment by lsloan** [comment]
Looks great!

:shipit:

File: server/data_provider_esb.rb [hunk 1]
```diff
@@ -27,87 +30,84 @@ def init_ESB(security_file, app_name)
...
+      ## Fix up unexpected values from ESB where there is no detail level data at all.  This can happen,
+      ## for example, a user has no term data at all.
+      ## Make these conditions separate so easy to take out when ESB returns expected values.
+      return WAPIResultWrapper.new(WAPI::SUCCESS, "replace nil value with empty array", []) if query_key_value.nil?
+      return WAPIResultWrapper.new(WAPI::SUCCESS, "replace empty string with empty array", []) if query_key_value.length == 0
+
+      parsed_value = parsed[query_key][detail_key]
+
+      # if there is a detail_key but no data that's an error.
+      raise "ESBInvalidData: input: #{result}" if parsed_value.nil?
```
Comment (zqian) [inline server/data_provider_esb.rb]: Is it OK for parsed_value to be empty string?

File: server/spec/test_WAPI_result_wrapper.rb [hunk 2]
```diff
@@ -76,6 +76,24 @@ def test_instance_full_value
     assert valid_json?(x), "wrapped result should be json"
   end
 
+  def test_value_of_nil
+    # if get back nil terms still have it parse ok.
+    w = WAPIResultWrapper.new("status", "msg", Hash.new['getMyRegTermsResponse']=nil)
+    refute_nil(w, "create wrapped object")
+    assert w.valid?, "check for valid wrapper object"
```
Comment (zqian) [inline server/spec/test_WAPI_result_wrapper.rb]: line 83-85 is same to line 92-94. Refact?

File: server/test-files/terms/empty-string.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (zqian) [inline server/test-files/terms/empty-string.json]: Did not see where this JSON file is used inside current diff

**Pull request merged**